src/services/langfuse.py

```diff
@@ -101,75 +97,73 @@ def masking_production_data(self, data: Any) -> Any:
             return data
         elif self.masking_mode == LangfuseMaskingModes.REDACTED:
             # If masking is set to REDACTED, return a placeholder string
-            return "REDACTED"
+            return REDACTED
+        elif self.masking_mode == LangfuseMaskingModes.PARTIAL:
+            return self._masking_mode_partial(data)
+        elif self.masking_mode == LangfuseMaskingModes.FILTERED:
+            return self._masking_mode_filtered(data)
 
-        # If masking is set to PARTIAL, return only the user input and resource information.
+        return REDACTED
+
+    def _masking_mode_partial(self, data: Any) -> Any:
+        """Return only the user input and resource information. Everything else is redacted."""
         if isinstance(data, GraphInput):
             output = "\n".join([str(msg.content) for msg in reversed(data.messages)])
             if output:
                 return self.data_scrubber.clean(output)
 
-        return "REDACTED"
-
-    async def get_daily_metrics(
-        self,
-        from_timestamp: str,
-        to_timestamp: str,
-        tags: Any = None,
-        user_id: Any = None,
-    ) -> MetricsResponse | None:
+        return REDACTED
+
+    def _masking_mode_filtered(self, data: Any) -> Any:  # noqa: C901
```

Comment:
```suggestion
    def _masking_mode_filtered(self, data: Any) -> Any:  # noqa: C901
```
The linter has a point about the complexity here. I assume it would be pretty hard to reduce the complexity, here?

Reply:
yes